### DISCUSSION TITLE
How to optimize for low RAM / raspberry pi

### DISCUSSION THREAD

**Discussion by joe-8888** [WayfireWM/wayfire#2181] [General]
Hi all,

First of all thank you to everyone for contributing and maintaining wayfire. 

I am using raspbian on my 1gb RAM Raspberry Pi 4 and noticed high RAM and CPU usage for wayfire especially when opening a web browser. All packages are up to date and mostly unchanged from default.

Is there anything I can remove / disable / adjust with my wayfire configuration to reduce resource use? I appreciate a Pi with 1gb RAM is very limited, but my use-case is also very limited: I don't need anything beyond the minimum to open a browser and access my router webportal and pihole webportal, via a RealVNC connection if that matters.

my **defaults.ini** file:
```
[core]
plugins = alpha animate autostart autostart-static command cube pixdecor expo fast-switcher fisheye grid idle invert move oswitch place resize switcher vswitch window-rules wm-actions wrot zoom winshadows
plugins_nogl = alpha animate autostart autostart-static command pixdecor expo fast-switcher grid idle move oswitch place resize switcher vswitch window-rules wm-actions wrot
xwayland = true
only_decorate_gtk = true

[expo]
toggle = 

[winshadows]
include_undecorated_views = true

[pixdecor]
always_decorate = app_id is "Kodi"

[grid]
slot_l = <ctrl><alt>KEY_LEFT
slot_r = <ctrl><alt>KEY_RIGHT
slot_c = <ctrl><alt>KEY_UP
restore = <ctrl><alt>KEY_DOWN
mouse_snap = false

[autostart]
autostart_wf_shell = false

[autostart-static]
autostart0 = wfrespawn wf-panel-pi
autostart1 = wfrespawn pcmanfm --desktop --profile LXDE-pi
autostart2 = lxsession-xdg-autostart

[input]
cursor_theme = PiXflat

[idle]
screensaver_timeout = -1
dpms_timeout = -1

```

**Comment by soreau** [comment]
Hi @joe-8888. I'm not sure why wayfire is using high cpu when running a web browser, but I would say, this is not common. There are a couple things you might try, however. 1) Remove any plugins from the [core] plugins list that you don't have a use for. 2) Install latest wayfire to /opt/wayfire/ by running the wf-install script. 2 will install the latest version of wayfire to /opt/wayfire/ by default. You can run it with /opt/wayfire/bin/startwayfire and if it doesn't help, simply remove /opt/wayfire/ directory to uninstall it. It will not affect your default wayfire installation in /usr/.

**Reply by ammen99** [reply]
I was also going to suggest disabling all plugins that you don't need. 

Also consider that if you are using raspberry-pi's own version it might be using the pixman renderer without any hardware acceleration. That will use a lot of CPU, there is no going around that. Have you looked at the log file from Wayfire? (Log is by default printed to stdout, but I don't know whether the version shipped in raspberry os doesn't automatically redirect it somewhere). It ought to say whether it is using OpenGL or Pixman renderers.
